example/calculate_pi/calculate_pi.cpp

```diff
@@ -149,18 +149,18 @@ TaskStatus ComputeArea(MeshBlock *pmb) {
   auto &coords = pmb->coords;
 
   ParArrayND<Real> &v = rc.Get("in_or_out").data;
-  const auto &radius = pmb->packages["calculate_pi"]->Param<Real>("radius");
 
-  Kokkos::deep_copy(pmb->exec_space, v.Get(0, 0, 0, 0, 0, 0), 0.0);
+  Real area;
   Kokkos::parallel_reduce(
       "calculate_pi compute area",
       Kokkos::MDRangePolicy<Kokkos::Rank<3>>(pmb->exec_space, {kb.s, jb.s, ib.s},
                                              {kb.e + 1, jb.e + 1, ib.e + 1},
                                              {1, 1, ib.e + 1 - ib.s}),
-      KOKKOS_LAMBDA(int k, int j, int i, Real &area) {
-        area += v(k, j, i) * coords.Area(parthenon::X3DIR, k, j, i);
+      KOKKOS_LAMBDA(int k, int j, int i, Real &larea) {
+        larea += v(k, j, i) * coords.Area(parthenon::X3DIR, k, j, i);
       },
-      Kokkos::Sum<Real, parthenon::DevMemSpace>(v.Get(0, 0, 0, 0, 0, 0)));
+      area);
+  Kokkos::deep_copy(pmb->exec_space, v.Get(0, 0, 0, 0, 0, 0), area);
```

Comment:
Oh, I intentionally didn't do this so that it wouldn't fence here. Why did this need to change? I'm having a hard time understanding what the issue was.

Reply:
There was no real "issue" (the result for the specific test didn't change), but conceptually I was confused by the original deep_copy to to `v` and then using `v` in the reduction loop (i.e., if `kb.s`, `jb.s`, and `ib,s` would be 0 then the original value stored in `v(0,0,0)` would be lost for the area summation (independent of he final reduction to the view).
Given that this should serve as an example, I think the current version is easier to follow.